src/pages/activities/list/List.component.js

```diff
@@ -124,17 +137,25 @@ const List = () => {
         </Row>
       </Styled.Filters>
 
-      <Row gutter={[32, 32]}>
-        {activities.map((activity, idx) => {
-          const props = {
-            key: idx,
-            ...activity,
-            image: commonUtils.getStrapiImgProps(activity.image),
-          };
-
-          return <Activity {...props} />;
-        })}
-      </Row>
+      {activities?.length ? (
+        <Row gutter={[32, 32]}>
+          {activities.map((activity, idx) => {
+            const props = {
+              key: idx,
+              ...activity,
+              image: commonUtils.getStrapiImgProps(activity.image),
+            };
+
+            return <Activity {...props} />;
+          })}
+        </Row>
+      ) : !activities ? (
+        <Styled.SpinContainer>
+          <Spin size="large" />
+        </Styled.SpinContainer>
+      ) : (
+        <Empty description={lang.empty} />
+      )}
```

Comment:
Display the loading spinner when the API response is not coming back. If there is no available data, then showing the Empty component with a proper explaination.